test/negative/invalid-vloada-half.ll

```diff
@@ -0,0 +1,61 @@
+; RUN: llvm-as %s -o %t.bc
+; RUN: not llvm-spirv %t.bc -o %t.spv 2>&1 | FileCheck %s
+
+; CHECK: InvalidModule: Invalid SPIR-V module: vloada_half should be of a half vector type
+
+; ModuleID = 'loada.bc'
+target datalayout = "e-p:64:64:64-i1:8:8-i8:8:8-i16:16:16-i32:32:32-i64:64:64-f32:32:32-f64:64:64-v16:16:16-v24:32:32-v32:32:32-v48:64:64-v64:64:64-v96:128:128-v128:128:128-v192:256:256-v256:256:256-v512:512:512-v1024:1024:1024"
+target triple = "spir64-unknown-unknown"
+
+; Function Attrs: noinline nounwind
+define spir_kernel void @test(half addrspace(1)* readonly %p, float addrspace(1)* %f) #0 {
+entry:
+  %p.addr = alloca half addrspace(1)*, align 8
+  %f.addr = alloca float addrspace(1)*, align 8
+  %i = alloca i64, align 8
+  store half addrspace(1)* %p, half addrspace(1)** %p.addr, align 8
+  store float addrspace(1)* %f, float addrspace(1)** %f.addr, align 8
+  %0 = call spir_func i64 @_Z13get_global_idj(i32 0) #1
+  %1 = insertelement <3 x i64> undef, i64 %0, i32 0
+  %2 = call spir_func i64 @_Z13get_global_idj(i32 1) #1
+  %3 = insertelement <3 x i64> %1, i64 %2, i32 1
+  %4 = call spir_func i64 @_Z13get_global_idj(i32 2) #1
+  %5 = insertelement <3 x i64> %3, i64 %4, i32 2
+  %call = extractelement <3 x i64> %5, i32 0
+  store i64 %call, i64* %i, align 8
```

Comment:
I suppose that this part is optional and may be dropped to reduce the test.
Also it would be nice to add check for non-half vector type.

Reply:
Done, thanks!